pkg/controller/migmigration/validation.go

```diff
@@ -170,7 +170,8 @@ func (r ReconcileMigMigration) validateFinalMigration(plan *migapi.MigPlan, migr
 			}
 		}
 	}
-	if hasCondition {
+	// Allow perform Rollback on finished Plan.
+	if hasCondition && !migration.Spec.Rollback {
```

Comment:
+1, I think this makes sense. We need to override this logic that blocks another migration from happening after a completed final migration in order to do rollback.